api/controller.go

```diff
@@ -227,6 +231,28 @@ func (c *Controller) updateDeployment(d service.Deploy, ddLast *service.DynamoDe
 				// don't update ecs service later
 				updateECSService = false
 			}
+			ps := ecs.Paramstore{}
+			if ps.IsEnabled() {
+				iam := ecs.IAM{}
+				thisNamespace, lastNamespace := d.EnvNamespace, ddLast.DeployData.EnvNamespace
+				if thisNamespace == "" {
+					thisNamespace = serviceName
+				}
+				if lastNamespace == "" {
+					lastNamespace = serviceName
+				}
+				if thisNamespace != lastNamespace {
+					controllerLogger.Debugf("Paramstore enabled, putting role: paramstore-%v", serviceName)
+					err = iam.DeleteRolePolicy("ecs-"+serviceName, "paramstore-"+lastNamespace)
+					if err != nil {
+						return err
+					}
+					err = iam.PutRolePolicy("ecs-"+serviceName, "paramstore-"+thisNamespace, ps.GetParamstoreIAMPolicy(thisNamespace))
```

Comment:
Yes, but the service technically only needs access when starting a task. So this should be safe deleting and reading, since the actual deploy is done a bit later.